test/quantization/pt2e/test_quantize_pt2e_qat.py

```diff
@@ -732,3 +732,78 @@ def test_mixing_qat_ptq(self):
         self.checkGraphModuleNodes(
             exported_model.graph_module, expected_node_occurrence=node_occurrence
         )
+
+    def test_mixing_qat_ptq_with_checkpointing(self):
+        import io
+
+        buffer = io.BytesIO()
+        example_inputs = (torch.randn(2, 3, 4, 4),)
+        model = TestQuantizeMixQATAndPTQ.QATPTQTestModule()
+        model.train()
+
+        self._prepare_qat_linears(model)
+
+        loss = torch.nn.MSELoss()
+        optimizer = torch.optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
+        for i in range(2):
+            after_prepare_result_pt2e = model(*example_inputs)
+            target = torch.rand_like(after_prepare_result_pt2e)
+            output = loss(after_prepare_result_pt2e, target)
+            output.backward()
+            optimizer.step()
+
+        state_dict_before = copy.deepcopy(model.state_dict())
+        torch.save(model.state_dict(), buffer)
+        buffer.seek(0)
+        model.load_state_dict(torch.load(buffer))
```

Comment:
nit: to simulate the real use case, this model should be initialized from the beginning I think

Reply:
what do you mean?

Reply:
oh I mean we should repeat L741-744 here before loading the state_dict to simulate the use case that people do checkpointing, and resume training a bit later

Reply:
Thanks for pointing this out. It does require running through prepare_qat step again.